versioned_docs/version-8.0/guides/update-guide/connectors/010-to-020.md

```diff
@@ -0,0 +1,173 @@
+---
+id: 010-to-020
+title: Update 0.1 to 0.2
+description: "Review which adjustments must be made to migrate from Connector SDK 0.1.x to 0.2.0."
+---
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+<span class="badge badge--primary">Intermediate</span>
+
+The following sections explain which adjustments must be made to migrate from
+[Connector SDK](/components/integration-framework/connectors/custom-built-connectors/connector-sdk.md)
+0.1.x to 0.2.0.
+
+:::caution
+
+Be aware that the update from 0.1 to 0.2 requires manual migration steps as described below.
+
+:::
+
+## Connector runtime
+
+With SDK version 0.2.0, we introduce the following structural changes:
+
+- Input validation and secret replacement move from writing imperative code to declaratively using annotations.
+- The Outbound aspect of APIs is more explicit. Classes have been moved to more explicit packages and have been renamed.
+
+### Declarative validation and secrets
+
+Input objects previously had to implement the `ConnectorInput` interface to participate in validation and secret replacement
+initiated from the `ConnectorContext` using its `validate` and `replaceSecrets` methods respectively.
+
+With version 0.2.0, we remove the imperative approach for validation and secret replacement from the SDK.
+Instead, you can use annotations to describe the constraints of input attributes and mark those that can contain
+secrets.
+
+These are two input objects written with the SDK version 0.1.x:
+
+```java
+import io.camunda.connector.api.ConnectorInput;
+import io.camunda.connector.api.SecretStore;
+import io.camunda.connector.api.Validator;
+
+record MyConnectorRequest(String message, Authentication authentication) implements ConnectorInput {
+  @Override
+  public void validateWith(final Validator validator) {
+    validator.require(message, "message");
+    validator.require(authentication, "authentication");
+    validateIfNotNull(authentication, validator);
+  }
+
+  @Override
+  public void replaceSecrets(final SecretStore secretStore) {
+    replaceSecretsIfNotNull(authentication, secretStore);
+  }
+}
+```
+
+```java
+import io.camunda.connector.api.ConnectorInput;
+import io.camunda.connector.api.SecretStore;
+import io.camunda.connector.api.Validator;
+
+record Authentication(String user, String token) implements ConnectorInput {
+  @Override
+  public void validateWith(final Validator validator) {
+    validator.require(user, "user");
+    validator.require(token, "token");
+    if (token != null && !(token.startsWith("xobx") || token.startsWith("secrets."))) {
+      validator.addErrorMessage("Token must start with \"xobx\" or be a secret");
+    }
+  }
+
+  @Override
+  public void replaceSecrets(final SecretStore secretStore) {
+    token = secretStore.replaceSecret(token);
```

Comment:
I didn't really work with java records but I thought, this shouldn't be possible as records components are immutable. Like, from [Java doc](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/Record.html):
> A record class is a shallowly immutable, transparent carrier for a fixed set of values, called the record components. The Java language provides concise syntax for declaring record classes, whereby the record components are declared in the record header. The list of record components declared in the record header form the record descriptor.

Reply:
Ehm, yeah, true 🤣
I will adjust the examples 😊

Reply:
Adjusted